src/vs/workbench/api/node/extHostTerminalService.ts

```diff
@@ -310,6 +311,11 @@ export class ExtHostTerminalService implements ExtHostTerminalServiceShape {
 		private _logService: ILogService,
 	) {
 		this._proxy = mainContext.getProxy(MainContext.MainThreadTerminalService);
+		this._extHostConfiguration.getConfigProvider().then((configProvider) => {
```

Comment:
We should pull this into a function and do it whenever the active workspace changes